compiler/luci/pass/src/EliminateDeadSubgraphPass.cpp

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/EliminateDeadSubgraphPass.h"
+
+#include <luci/IR/CircleNodes.h>
+
+#include <unordered_set>
+#include <deque>
+
+namespace luci
+{
+
+namespace
+{
+
+// Go through the current graph and check all other graphs reachable from it and save it.
+// Note: The main idea for finding achievable graphs is that we can reach other graphs only
+// from some operations (see the list below) and we check the graph indexes from these operations.
+void checkGraph(loco::Graph *current_graph, std::deque<size_t> &reachable_graphs_indexes_q)
+{
+  assert(current_graph != nullptr);
+
+  // 1 - Obtain all active nodes in current graph
+  // 2 - Go through all active nodes and check its types
+  // 3 - If it is possible to reach another graph from the current operation (see the list below),
+  //     then add the graph numbers to our queue
+
+  // 1 - Obtain all active nodes in current graph
+  // Let's enumerate nodes required to compute output nodes
+  auto active_nodes = loco::active_nodes(loco::output_nodes(current_graph));
+
+  // 2 - Go through all active nodes and check its types
+  // Nodes from we can obtain different subgraph:
+  // While, If, ...
+  // TODO: check all nodes which can be used to reach different subgraph
+  for (auto &node : active_nodes)
+  {
+    auto *circle_node = dynamic_cast<luci::CircleNode *>(node);
+    assert(circle_node != nullptr);
+
+    switch (circle_node->opcode())
+    {
+      case CircleOpcode::WHILE:
+      {
+        auto *while_node = dynamic_cast<luci::CircleWhile *>(circle_node);
+        assert(while_node != nullptr);
+        // Get body and cond graph indexes
+        int32_t body_graph_index = while_node->body_branch();
+        int32_t cond_graph_index = while_node->cond_branch();
+        assert(body_graph_index >= 0);
+        assert(cond_graph_index >= 0);
+        // Add indexes into queue
+        reachable_graphs_indexes_q.push_back(size_t(body_graph_index));
```

Comment:
Q) why use `size_t(body_graph_index)` ? this will always be 4

Reply:
Done